src/Symfony/Component/Console/DependencyInjection/AddConsoleCommandPass.php

```diff
@@ -55,7 +55,7 @@ public function process(ContainerBuilder $container)
 
             $commandId = 'console.command.'.strtolower(str_replace('\\', '_', $class));
 
-            if (!isset($tags[0]['command'])) {
+            if (!isset($tags[0]['command']) && !$commandName = $class::getDefaultName()) {
```

Comment:
so, what should happen when $commandName === '' ? Registering the empty string is a no-go to me. As is LGTM.